### ISSUE TITLE
Add `list` subcommand

### ISSUE THREAD

**Issue by gdemonet** [scality/changelog-binder#4]
The command-line tool defined in #3 should first have sufficient capabilities to extract the list of changelog fragments from Git history.
For this purpose, we want to add a `list` subcommand, with the following behaviour:
- `list --in X.Y.Z` would return all fragments included **before** the X.Y.Z tag but **after** the most recent tag that comes before it (in terms of Git history)
- `list --from A.B.C --to D.E.F` would return fragments added **after** A.B.C and **before** D.E.F
- `list --from X.Y.Z` would return fragments added **after** X.Y.Z up to the current HEAD
- `list` would return fragments added after the latest tag

**Comment by NicolasT** [comment]
I was looking at implementing this, and it occurred to me we could do this somewhat differently, more alike how `git diff` etc. work. Basically:

- `changelog-binder list` is as you describe without options (somewhat like `git diff` in a tree, diff from last commit to working tree)
- `changelog-binder list X.Y.Z` would be as-if `--from X.Y.Z` is used (somewhat like `git diff ref1`, diff between given ref and working tree)
- `changelog-binder list A.B.C...D.E.F` would be as-if `--from A.B.C --to D.E.F` is used (somewhat like `git diff ref1...ref2`, diff between ref1 and ref2)

The `--in` variant could be something like `changelog-binder list X.Y.Z^...X.Y.Z`

Basically, we treat version strings as Git treats commit refs.
